### ISSUE TITLE
Runner Configuration via terraform cr

### ISSUE THREAD

**Issue by oliverbaehler** [weaveworks/tf-controller#166]
Hi, really like the project!

One thing I noticed (I am missing) is the possibility to configure a terraform runner. This could come in handy if I eg. need to set resources or might want to extend to environment of the runner. I could imagine this would look something like this:

```
apiVersion: infra.contrib.fluxcd.io/v1alpha1
kind: Terraform
metadata:
  name: helloworld
  namespace: flux-system
spec:
  approvePlan: "auto"
  path: ./
  interval: "1m"
  sourceRef:
    kind: GitRepository
    name: tf-catalog
    namespace: flux-system

   # that assumes you can patch the entire pod spec of the runner. I don't how possible that's gonna be, but i thought it might also be interesting to adjust the metadata of the pod.
    runner:
      metadata:
        labels:
          company.label/type: terraform
        annotations:
           kubernetes.io/psp: custom-psp # I know.
     spec:
       containers:
       - resources:
            limits:
              cpu: 1000m
              memory: 1Gi
            requests:
              cpu: 200m
             memory: 64Mi
```

As alternative I could approach something like this via kyverno selecting pods with the `infra.contrib.fluxcd.io/terraform` and mutate them. But I first wanted to check with you guys, what you think about this.

Thanks

**Comment by chanwit** [comment]
Yeah, this is making sense. Thank you for the suggestion, @oliverbaehler !!

**Comment by chanwit** [comment]
Changing `runner` to `runnerPod` and it makes a good spec. Thank you @oliverbaehler !

```
apiVersion: infra.contrib.fluxcd.io/v1alpha1
kind: Terraform
metadata:
  name: helloworld
  namespace: flux-system
spec:
  approvePlan: "auto"
  path: ./
  interval: "1m"
  sourceRef:
    kind: GitRepository
    name: tf-catalog
    namespace: flux-system
  runnerPod:
    metadata:
      labels:
        company.label/type: terraform
      annotations:
        kubernetes.io/psp: custom-psp
    spec:
      resources:
        limits:
          cpu: 1000m
          memory: 1Gi
        requests:
          cpu: 200m
          memory: 64Mi
```

**Comment by tomhuang12** [comment]
I think I could put together a quick PR to customize runner pod's metadata. We should probably put more thoughts into customizing the `spec` section since there are some required [configurations](https://github.com/weaveworks/tf-controller/blob/main/controllers/terraform_controller.go#L1972-L1998) needed to be able to run it correctly with the controller and overriding them could cause incorrect behavior. Maybe we only allow customizing only a subset of the `spec`?

**Comment by chanwit** [comment]
Hi @oliverbaehler 
we still need documentation for it, but the implementation already landed in v0.9.5-rc.3.
